feat(fe): warn on confusing operator precedence of & and >> (#1093)
quick-lint/quick-lint-js: docs/errors/E0715.md

# E0715: unintuitive operator precedence when using & and << or >>

> Javascript's operator precedence can cause confusing situations and bugs when it

Reviewer: Nit: It's spelled JavaScript, not Javascript:
```suggestion
JavaScript's operator precedence can cause confusing situations and bugs when it
```

comes to bitshifting with a bitwise and.
```js
let a_bitwise_variable = another_variable & 0x1 << 0x2
```
The code above looks like it evaluates `another_variable & 0x1` first,
but it instead evaluates `0x1 << 0x2` first. 

> If this is intended behavior, it may be easier to read if you write:

Reviewer: Nit: Explain in words what is being done:
```suggestion
If this is intended behavior, it may be easier to read if you write parentheses:
```

```js
another_variable & (0x1 << 0x2)
```

> Otherwise, to get the order of operations correct, do:

Reviewer: Nit: Explain in words what is being done:
```suggestion
Otherwise, to get the order of operations correct, add parentheses around the operands of `&`:
```
